cmd/protoc-gen-jsonshim/jsonshim/plugin.go

```diff
@@ -67,14 +67,19 @@ func (p *Plugin) Generate(file *generator.FileDescriptor) {
 	unmarshalerName := generator.FileName(file) + "Unmarshaler"
 	for _, message := range file.Messages() {
 		// check to make sure something was generated for this type
-		if !gogoproto.HasTypeDecl(file.FileDescriptorProto, message.DescriptorProto) || len(message.GetOneofDecl()) == 0 {
+		if !gogoproto.HasTypeDecl(file.FileDescriptorProto, message.DescriptorProto) {
+			continue
+		}
+
+		// skip maps in protos
+		if message.Options != nil && message.Options.GetMapEntry() {
```

Comment:
Why are we skipping maps here?

Reply:
we don't need to create marshal methods for go's map type, `encoding/json` handles it.